src/infra/core/tests/tests/test_compact_service_impl.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::assert_matches::assert_matches;
+use std::sync::Arc;
+
+use dill::Component;
+use domain::compact_service::{CompactError, CompactService, NullCompactionMultiListener};
+use event_bus::EventBus;
+use futures::TryStreamExt;
+use kamu::domain::*;
+use kamu::testing::{DatasetTestHelper, MetadataFactory};
+use kamu::*;
+use kamu_core::{auth, CurrentAccountSubject};
+use opendatafabric::*;
+const FILE_DATA_ARRAY_SIZE: usize = 32;
+
+#[tokio::test]
+async fn test_dataset_compact() {
```

Comment:
Must be tested with S3/Minio container as well.

Reply:
LFS vs S3 is a separate topic from local vs remote.
Regarding remote, if any side of the local or remote is compacted separately, it's no longer possible to sync. Although, as far as I understand, if compacting with the same limits is applied to both copies, this would produce identical results.

Reply:
On the other hand, it's the same for GIT. If someone force-pushed history rewrite change like `git rebase`, the other side has to use `git pull --rebase`, a normal procedure fails due to history divergence. We have "-force" switch in `kamu pull` and `kamu push` too.